pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -149,36 +149,28 @@
   // Only instantiated if a task is actually scheduled.
   private volatile Timer cancelTimer = null;
 
-  // In general, it is not expected that connection would be used in multiple threads concurrently
-  // However, if client performs cleanup of statements via finalizers/phantomreferences, then it
-  // might
-  // result in AbstractJdbc2Statement.close to be called concurrently with active operation on
-  // connection
-  // Thus we synchronize access to avoid java.util.ConcurrentModificationException
   private final LruCache<Object, CachedQuery> statementCache;
+  private final LruCache<String, DatabaseMetadataCache> metadataCache;
 
   private boolean reWriteBatchedInserts = false;
 
   CachedQuery borrowQuery(String sql, boolean isCallable) throws SQLException {
     Object key = isCallable ? new CallableQueryKey(sql) : sql;
-    // Synchronized to fix #368
-    synchronized (statementCache) {
-      return statementCache.borrow(key);
-    }
+    return statementCache.borrow(key);
```

Comment:
`syncrhonized` is here for a good reason.
Please check https://github.com/Gordiychuk/pgjdbc/commit/40bcc011e77a710be0cccc7e34f4270375db6ebc and https://github.com/pgjdbc/pgjdbc/issues/368#issuecomment-152715832.
Are you still sure that synchronization is not required?